### PR TITLE
[master < T082-MAGE] Dynamic betweenness centrality documentation

### PR DESCRIPTION
**Pull request by antepusic**
Documentation for MAGE’s dynamic betweenness centrality query module (https://github.com/memgraph/mage/pull/127).

- [x] Query module documentation
- [x] Algorithm description

### CODE REVIEW

File: mage/algorithms/dynamic-graph-analytics/betweenness-centrality-online-algorithm.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+measuring betweenness centrality helps identify bottlenecks and choke points, as well as monitor
+how traffic is redistributed as nodes or connections between them become open or close.
+This can help with applications such as shipping/delivery routes or urban traffic grid optimization
+on both cost and benefit fronts.
+
+[![Bioinformatics](https://img.shields.io/badge/Bioinformatics-Application-8A477F?style=for-the-badge)](/use-cases/bioinformatics.md)
+
+[**Protein-protein interaction
+networks**](https://www.ebi.ac.uk/training/online/courses/network-analysis-of-protein-interaction-data-an-introduction/protein-protein-interaction-networks/)
+were created to get a better insight into how proteins interact with each other.
```
Comment (antepusic) [inline mage/algorithms/dynamic-graph-analytics/betweenness-centrality-online-algorithm.md:79]: Are dynamic protein-protein interaction networks used in practice? If not, I'm ready to remove this.

**Review by g-despot — APPROVED** [review]

**Review by antepusic — COMMENTED** [review]
@g-despot Thanks!

File: mage/algorithms/dynamic-graph-analytics/betweenness-centrality-online-algorithm.md (hunk 1 above)
Comment (jmatak) [reply mage/algorithms/dynamic-graph-analytics/betweenness-centrality-online-algorithm.md:88]: I think they are used. We got a hint from one developer that is using Memgraph.

Comment (antepusic) [reply mage/algorithms/dynamic-graph-analytics/betweenness-centrality-online-algorithm.md:88]: Thanks! Could you share what they said with me?

File: mage/algorithms/dynamic-graph-analytics/betweenness-centrality-online-algorithm.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+centrality](https://en.wikipedia.org/wiki/Betweenness_centrality) is based on
+shortest paths: the shortest path between two nodes is the one consisting of the
+fewest edges, or in case of weighted graphs, the one with the smallest total
+edge weight. A node’s betweenness centrality is defined as the share of all
+shortest paths in the graph that run through it.
+
+MAGE includes a _fully dynamic_ betweenness centrality computation tool that
+implements the
+[iCentral](https://repository.kaust.edu.sa/bitstream/handle/10754/625935/08070346.pdf)
+[^1] algorithm. iCentral saves up on computation in two ways: it singles out the
```
Comment (jmatak) [inline mage/algorithms/dynamic-graph-analytics/betweenness-centrality-online-algorithm.md:24]: Write the reference below and name the author

File: mage/algorithms/dynamic-graph-analytics/betweenness-centrality-online-algorithm.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+> After the node in the center is erased, shortest paths run through other nodes
+> and their betweenness centrality scores increase. A larger node size signifies
+> a higher betweenness centrality score.
+
+## Materials
+
+### Implementation
+
+[![Betweenness Centrality
```
Comment (jmatak) [inline mage/algorithms/dynamic-graph-analytics/betweenness-centrality-online-algorithm.md:46]: Fix badge

File: mage/algorithms/dynamic-graph-analytics/betweenness-centrality-online-algorithm.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+Online](https://img.shields.io/badge/Betweenness_Centrality-Online-Implementation-FB6E00?style=for-the-badge&logo=github&logoColor=white)](https://github.com/memgraph/mage/blob/main/cpp/betweenness_centrality_module/betweenness_centrality_module.cpp)
+
+[![Betweenness Centrality
```
Comment (jmatak) [inline mage/algorithms/dynamic-graph-analytics/betweenness-centrality-online-algorithm.md:49]: Fix badge

File: mage/query-modules/cpp/betweenness_centrality_online.md [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+      backgroundColor: color,
+      borderRadius: '2px',
+      color: '#fff',
+      padding: '0.2rem',
+    }}>
+{children}
+</span>
+);
+
+[![docs-source](https://img.shields.io/badge/source-betweenness_centrality-online-FB6E00?logo=github&style=for-the-badge)](https://github.com/memgraph/mage/blob/main/cpp/betweenness_centrality_online_module/betweenness_centrality_online_module.cpp)
```
Comment (jmatak) [inline mage/query-modules/cpp/betweenness_centrality_online.md:22]: Fix badge not found error

Comment (antepusic) [reply mage/query-modules/cpp/betweenness_centrality_online.md:22]: Fixed

File: mage/query-modules/cpp/betweenness_centrality_online.md [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+  </TabItem>
+  <TabItem value="result">
+
+```plaintext
+┌─────────────────────────┬─────────────────────────┐
+│ node_id                 │ betweenness_centrality  │
+├─────────────────────────┼─────────────────────────┤
+│ 0                       │ 0                       │
+│ 1                       │ 0                       │
+│ 2                       │ 0.6                     │
```
[the remark this answers is not in the document]
Comment (antepusic) [reply mage/query-modules/cpp/betweenness_centrality_online.md:217]: Normalization just means that the raw betweenness centrality scores are divided by the factor of (N - 1)(N - 2)/2 (or just (N - 1)(N - 2) for directed graphs) and not by the highest score(s).

File: mage/algorithms/dynamic-graph-analytics/betweenness-centrality-online-algorithm.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+**Social networks** are a quintessential example of rapidly evolving graphs –
+people step in and out of contact every day, and even a single change can
+significantly alter the network of relationships. Furthermore, individual users’
+activities often come in _bursts_ of many interactions in short order. With
+real-time network-aware applications (e.g. finding influential persons, trend
+analysis), being able to instantly reflect graph changes into centrality values
+is crucial.
+
+[![Logistics](https://img.shields.io/badge/Logistics-Application-8A477F?style=for-the-badge)](/use-cases/transportation.md)
```
[the remark this answers is not in the document]
Comment (antepusic) [reply mage/algorithms/dynamic-graph-analytics/betweenness-centrality-online-algorithm.md:66]: Fixed!

File: mage/algorithms/dynamic-graph-analytics/betweenness-centrality-online-algorithm.md (hunk 2 above)
Comment (antepusic) [reply mage/algorithms/dynamic-graph-analytics/betweenness-centrality-online-algorithm.md]: I’ve added the authors’ names. The reference is at the bottom of **Abstract**, is that OK?

File: mage/query-modules/cpp/betweenness_centrality_online.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (jmatak) [reply mage/query-modules/cpp/betweenness_centrality_online.md:217]: You are correct, similar applies to the static betweenness centrality algorithm. Thanks for explaining!

File: mage/algorithms/dynamic-graph-analytics/betweenness-centrality-online-algorithm.md (hunk 2 above)
Comment (jmatak) [reply mage/algorithms/dynamic-graph-analytics/betweenness-centrality-online-algorithm.md]: Yes!

File: mage/algorithms/dynamic-graph-analytics/betweenness-centrality-online-algorithm.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+[![Betweenness
+Centrality Online](https://img.shields.io/badge/Betweenness_Centrality_Online-Documentation-FCC624?style=for-the-badge&logo=cplusplus&logoColor=white)](/mage/query-modules/cpp/betweenness-centrality-online)
+
+Dynamic Betweenness Centrality is implemented as part of the
+[**MAGE**](https://github.com/memgraph/mage) project. Be sure to check it out at
+the link above. :point_up:
+
+### Blog posts
+
+(to be written)
```
Comment (jmatak) [inline mage/algorithms/dynamic-graph-analytics/betweenness-centrality-online-algorithm.md:49]: We shouldn't merge until this is resolved

**Review by jmatak — CHANGES_REQUESTED** [review]
Add blog post link once available and we are done :tada:

File: mage/query-modules/cpp/betweenness_centrality_online.md [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+As there is a total of four complex obligatory parameters, setting the
+parameters by hand might be cumbersome. The recommended use of this method is to
+call it within a [trigger](/memgraph/reference-guide/triggers), making sure
+beforehand that all [predefined
+variables](/memgraph/reference-guide/triggers/#predefined-variables) are
+available:
+
+```cypher
+CREATE TRIGGER sample_trigger BEFORE COMMIT
+EXECUTE CALL community_detection_online.update(createdVertices, createdEdges, deletedVertices, deletedEdges, normalize, threads) YIELD *;
```
Comment (jmatak) [inline mage/query-modules/cpp/betweenness_centrality_online.md:147]: betweenness_centrality

File: mage/query-modules/cpp/betweenness_centrality_online.md [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+```
+
+Communities calculated by `update()` are accessible by subsequently calling
+`get()`:
+
+```cypher
+CALL betweenness_centrality.get()
```
Comment (jmatak) [inline mage/query-modules/cpp/betweenness_centrality_online.md:154]: betweenness_centrality_online

Comment (antepusic) [inline mage/query-modules/cpp/betweenness_centrality_online.md:154]:
```suggestion
CALL betweenness_centrality_online.get()
```

File: mage/query-modules/cpp/betweenness_centrality_online.md (hunk 9 above)
Comment (antepusic) [inline mage/query-modules/cpp/betweenness_centrality_online.md:147]:
```suggestion
EXECUTE CALL betweenness_centrality_online.update(createdVertices, createdEdges, deletedVertices, deletedEdges, normalize, threads) YIELD *;
```

**Review by g-despot — APPROVED** [review]

**Pull request merged**